deploy/chart/templates/rbac-namespace.yaml

```diff
@@ -0,0 +1,28 @@
+{{- if and .Values.rbac.enabled (gt (len .Values.rbac.namespaces) 0) -}}
+{{- range $.Values.rbac.namespaces }}
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: "{{ include "secret-transform.name" $ }}"
+  namespace: {{ . | quote }}
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["list", "watch", "update"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: "{{ include "secret-transform.name" $ }}"
+  namespace: {{ . | quote }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: "{{ include "secret-transform.name" $ }}"
+subjects:
+  - kind: ServiceAccount
+    name: {{ include "secret-transform.name" $ }}
+    namespace: {{ $.Release.Namespace }}
```

Comment:
I think secret-transform lists secrets using `client.Secrets()` which requires the cluster-wide permission to list secrets.

I'm not sure but maybe  secret-transform needs to be changed so that only the given namespaces are being listed, e.g. with some flag `--only-namespaces`, wdyt?

Reply:
Thanks @maelvls - I had this thought last night too. I only really focused on the helm change... but I will do my best to take a look at potential changes needed in the code too.